### ISSUE TITLE
run-windows broken on versions < 0.70 when VS2022 is installed

### ISSUE THREAD

**Issue by marlenecota** [microsoft/react-native-windows#10917]
### Problem Description

RNW versions < 0.70 don't build when using VS2022.

```shell
X:\rnw69\node_modules\react-native-windows\Microsoft.ReactNative\Utils\Helpers.h(21,8): error C2187: syntax error: 'typename' was unexpected here (compiling source file Modules\AppStateModule.cpp) [X:\
       rnw69\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]
```

### Steps To Reproduce

1. Install Visual Studio 2022
2. npx react-native init rnw69 --template react-native@^0.69
3. cd rnw69
4. npx react-native-windows-init --overwrite
5. npx react-native run-windows

### CLI version

8.0.6

### Environment

```markdown
info Fetching system and libraries information...
System:
    OS: Windows 10 10.0.25246
    CPU: (8) x64 Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
    Memory: 39.58 GB / 63.85 GB
  Binaries:
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0, 10.0.22000.0, 10.0.22621.0
  IDEs:
    Android Studio: AI-213.7172.25.2113.9014738
    Visual Studio: 17.4.33110.190 (Visual Studio Enterprise 2022), 16.11.33027.164 (Visual Studio Enterprise 2019)
  Languages:
    Java: 11.0.16.1 - C:\Program Files\Eclipse Adoptium\jdk-11.0.16.101-hotspot\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.7 => 0.69.7
    react-native-windows: 0.69.13 => 0.69.13
  npmGlobalPackages:
    *react-native*: Not Found
```

### Snack, code example, screenshot, or link to a repository

![image](https://user-images.githubusercontent.com/1422161/203197936-93b97af4-13c6-41b9-a9f0-8413bc94c72d.png)

**Comment by jonthysell** [comment]
TLDR: Our CLI doesn't actually searching for the correct version of VS, instead accepting *any* version of VS with the right components installed, an unfortunately, that means potentially trying to use unsupported, broken VS versions.

The problem is three-fold:

1. The CLI was never looking a version range we specified, because we weren't passing the minimum version param properly.
2. Even if it was passed properly, we weren't flooring our max version, giving ranges like `[16.7,17.7)` instead of `[16.7,17.0)` which would allow the CLI to select a version of VS 2022 even though only VS 2019 is supported on stable released <= 0.70.
3. While VS 2022 17.3 happened to work building RNW < 0.70, the newly released VS 2022 17.4 has breaking compiler changes.

**Comment by jonthysell** [comment]
Here's where the function was changed from `version` to `minVersion` back before 0.64. https://github.com/microsoft/react-native-windows/commit/e1821b205f26bd1f05c466b03f45b61ec7a3bb5e#diff-de26670b469d56f222f4a8a9dcbd4561679822307eef00bd5e2f9a76710a3efbR88
